### PR TITLE
feat(layer/dropdown/combo-box): update stories to Storybook v7

### LINKED ISSUE #11302: [carbon-web-components]: combo-box Storybook 7 update

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)

Closes #11302 #11303 #11300

### Description
This PR updates the stories of:
* Layer
* Dropdown
* Combo-box

As well as including `with-layer` helper component to be included in the `preview.js` file for Storybook configs.

### Changelog

**Changed**

- layer, dropdown, and combo-box stories to storybook v7

### CODE REVIEW

File: packages/carbon-web-components/src/components/combo-box/combo-box.stories.ts [hunk 1]
```diff
@@ -55,8 +52,79 @@ const sizes = {
   [`Large size (${DROPDOWN_SIZE.LARGE})`]: DROPDOWN_SIZE.LARGE,
 };
 
-export const Default = () => {
-  return html`
+const defaultArgs = {
+  direction: 'bottom',
```
Comment (m4olivei) [inline packages/carbon-web-components/src/components/combo-box/combo-box.stories.ts:56]: Should this use the constant? `DROPDOWN_DIRECTION.BOTTOM`?

File: packages/carbon-web-components/src/components/layer/layer.stories.ts [hunk 2]
```diff
@@ -70,37 +79,30 @@ export const UseLayer = () => {
...
 
-UseLayer.storyName = 'useLayer';
-
-export const Playground = (args) => {
-  const { level } = args?.[`${prefix}-layer-playground`] ?? {};
-  return html`
+export const Playground = {
+  args: defaultArgs,
+  argTypes: controls,
+  render: ({level}) => html`
```
Comment (m4olivei) [inline packages/carbon-web-components/src/components/layer/layer.stories.ts:89]: Here again, should be be bringing in the parameters? eg.
```suggestion
  parameters: {
    percy: {
      skip: true,
    },
  },
  render: ({level}) => html`
```

**Review by m4olivei — COMMENTED** [review]
Lots of small things. My intent here is to get on the same page so we have consistency across the components, since we're going through them all anyway. Sorry for the noise.

File: packages/carbon-web-components/src/components/dropdown/dropdown.stories.ts [hunk 3]
```diff
@@ -185,64 +264,26 @@ export const Playground = (args) => {
...
-      ),
-    }),
-  },
+  `
 };
 
-export const skeleton = () =>
-  html` <cds-dropdown-skeleton></cds-dropdown-skeleton> `;
+export const skeleton = {
+  render: () => html` <cds-dropdown-skeleton></cds-dropdown-skeleton> `
```
[the remark this answers is not in the document]
Comment (IgnacioBecerra) [reply packages/carbon-web-components/src/components/dropdown/dropdown.stories.ts:271]: I removed them due to some talk about using a percy alternative in the future, but you're right it should remain until then.

File: packages/carbon-web-components/src/components/layer/layer.stories.ts [hunk 4]
```diff
@@ -34,33 +44,32 @@ export const Default = () => {
...
-  `;
+  `
 };
 
-Default.storyName = 'Default';
-
-export const CustomLevel = () => {
-  return html`
+export const CustomLevel = {
+  name: 'Custom level',
```
[the remark this answers is not in the document]
Comment (IgnacioBecerra) [reply packages/carbon-web-components/src/components/layer/layer.stories.ts:51]: We've actually gone back and forth between all capitalized vs only the first letter options (as seen in the Dotcom storybook), so I will ask our designers this question to see what we should do about the CWC storybook. I'll remove the `name` property for now and once we have a consensus we can change to the desired one

**Review by kennylam — APPROVED** [review]
LGTM!

**Comment by kennylam** [comment]
@m4olivei Thanks for the thorough review. Completely agree on consistency across stories. Let me see if I can get the changes from https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/11248 into this branch to catch linter issues during commit.

**Review by m4olivei — APPROVED** [review]
This looks great!

Would especially be nice to get this one in, b/c many stories depend on the `<cds-layer>` component. For instance @RichKummer and I noticed there were issues with the layers in #11312.

**Comment by m4olivei** [comment]
> This looks great!
> 
> Would especially be nice to get this one in, b/c many stories depend on the `<cds-layer>` component. For instance @RichKummer and I noticed there were issues with the layers in #11312.

Scratch this. I was misunderstanding what was needed in other component stories to get layer's fully functioning. There were missing imports in some `*.stories.ts` files.

**Pull request merged**